From 39447b6a5799a160eae452db920fd0735a78638b Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Mon, 7 Jul 2014 18:24:48 +0200
Subject: [PATCH 24/31] BUG/MINOR: listener: set the listener's fd to -1 after
 deletion

This is currently harmless, but when stopping a listener, its fd is
closed but not set to -1, so it is not possible to re-open it again.
Currently this has no impact but can have after the abstract sockets
are modified to perform a complete close on soft-reload.

The fix can be backported to 1.5 and may even apply to 1.4 (protocols.c).
---
 src/listener.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/listener.c b/src/listener.c
index ec3a39b..a82ce81 100644
--- a/src/listener.c
+++ b/src/listener.c
@@ -236,6 +236,7 @@ int unbind_listener(struct listener *listener)
 
 	if (listener->state >= LI_PAUSED) {
 		fd_delete(listener->fd);
+		listener->fd = -1;
 		listener->state = LI_ASSIGNED;
 	}
 	return ERR_NONE;
-- 
1.8.5.5

